### ISSUE TITLE
Could not find a declaration file for module '@tailwindcss/<plugin>'.

### ISSUE THREAD

**Issue by acalvino4** [tailwindlabs/tailwindcss#9206]
**What version of Tailwind CSS are you using?**

3.1.4

**What build tool (or framework if it abstracts the build tool) are you using?**

Vite 3.0

**What version of Node.js are you using?**

16.16.0

**What browser are you using?**

N/A

**What operating system are you using?**

macOS 12.4

**Reproduction URL**

https://play.tailwindcss.com/jin96evF9U?file=config

Unfortunately I cannot reproduce my issue on Tailwind Play because it is specifically a typescript config issue and as far as I can tell, Tailwind play does not, and cannot, use a ts config file, nor an esm module config. (As far as I know, tw doesn't directly support esm modules, but I have had it working on a few projects with the workaround noted [here](https://github.com/tailwindlabs/tailwindcss/discussions/5858), and either way this issue is more about the typescript issue described below)

**Describe your issue**

Now that Typescript [is supported](https://tailwindcss.com/blog/tailwindcss-v3-1#first-party-type-script-types), I'm trying to use it for my config. When trying to import first-party plugins
```ts
import type {Config} from 'tailwindcss';
import typography from '@tailwindcss/typography';
import aspectRatio from '@tailwindcss/aspect-ratio';
import forms from '@tailwindcss/forms';
import lineClamp from '@tailwindcss/line-clamp';

const twconfig: Config = {
        ...
	plugins: [typography, aspectRatio, forms, lineClamp],
};

export default twconfig;
```
I get the error `Could not find a declaration file for module '@tailwindcss/typography'. ` or similar for each plugin, though for some reason the `forms` import doesn't throw that error.
<img width="701" alt="image" src="https://user-images.githubusercontent.com/43254861/187253680-4feabe0c-4fd0-4377-a99e-44f620c513f8.png">

**Comment by Lee6546** [comment]
Hello, @acalvino4 
check this thread:
https://github.com/tailwindlabs/tailwindcss-forms/issues/18#issuecomment-735448249

**Comment by thecrypticace** [comment]
I've published types files for the plugins. Thanks for the reminder. We did this for forms but apparently spaced on the other three plugins. :)

There's still an issue where our types think the plugins aren't allowed. Afaik this is a problem with typescript not picking up the right overload and we haven't spent the time to dive into exactly why this happens but there is an existing issue open for this and we will track that problem there: https://github.com/tailwindlabs/tailwindcss/issues/8566
